app/services/process_feed_service.rb

```diff
@@ -163,13 +166,22 @@ def find_or_create_conversation(xml)
       Conversation.find_by(uri: uri) || Conversation.create!(uri: uri)
     end
 
-    def find_status(uri)
-      if TagManager.instance.local_id?(uri)
+    def find_status_by_uri(uri, account)
+      if account.local?
+        local_id = TagManager.instance.unique_tag_to_local_id(uri, 'Status')
+        return Status.find_by(id: local_id)
+      end
+
+      Status.find_by(uri: uri, account: @account)
```

Comment:
In the database, the URI has a unique index that is not scoped to account_id, so this does not prevent collisions